### PR TITLE
misc: remove sha1 from tests, remove testhooks, minor cleanups

### PR DESCRIPTION
**Pull request by MichaHoffmann**
* removed sha1 hashes from tests
* removed internal/testhooks
* use `runtime.NumCPU` as default reindex maxprocs
* removed code that disables ooo indexing from tests ( since its not used anymore )

### CODE REVIEW

File: pkg/index/index.go [hunk 1]
```diff
@@ -422,26 +423,7 @@ func (x *Index) isEmpty() bool {
...
-	reindexMaxProcs.v = n
-}
-
-// ReindexMaxProcs returns the maximum number of concurrent goroutines that are
-// used during reindexing.
-func ReindexMaxProcs() int {
-	reindexMaxProcs.RLock()
-	defer reindexMaxProcs.RUnlock()
-	return reindexMaxProcs.v
-}
```
Comment (bradfitz) [inline pkg/index/index.go:444]: This seems unrelated?

Comment (bradfitz) [reply pkg/index/index.go:444]: Keep the setter/getter and just change the default of 4 to NumCPU?

Comment (MichaHoffmann) [reply pkg/index/index.go]: :heavy_check_mark:

**Review by bradfitz — APPROVED** [review]
